include/ccf/historical_queries_adapter.h

```diff
@@ -52,6 +53,7 @@ namespace ccf::historical
   ccf::endpoints::EndpointFunction adapter_v2(
     const HandleHistoricalQuery& f,
     AbstractStateCache& state_cache,
+    NetworkIdentitySubsystemInterface& network_identity_subsystem,
     const CheckHistoricalTxStatus& available,
     const TxIDExtractor& extractor = txid_from_header);
```

Comment:
We shouldn't add a new argument and still calls this `_v2`. Instead we should add a `_v3`, which probably takes an `AbstractNodeContext` rather than a `StateCache` _or_ a `IdentitySubsystem`, and gets both (+ whatever it needs in future, maybe) from the `AbstractNodeContext`. For the implementations, we keep as much common as we can, and we deprecate `_v2` if that's too hard/it can no longer be provided.

Reply:
That's a very good point, will add that immediately.